test/Generator/EnumGeneratorTest.php

```diff
@@ -0,0 +1,207 @@
+     *      }|array{
+     *          name: non-empty-string,
+     *          backedCases: array{
+     *              type: 'int'|'string',
+     *              cases: array<non-empty-string, int|non-empty-string>,
+     *          },
+     *      },
+     *      1: non-empty-string
+     * }>
+     */
+    public function validOptionSpecifications(): iterable
+    {
+        yield 'pure enum without namespace' => [
+            [
+                'name'      => 'Suit',
+                'pureCases' => ['Hearts', 'Diamonds', 'Clubs', 'Spades'],
+            ],
+            <<<CODE
+                enum Suit {
+                    case Hearts;
+                    case Diamonds;
+                    case Clubs;
+                    case Spades;
+                }
+                
+                CODE,
+        ];
+
+        yield 'pure enum with namespace' => [
+            [
+                'name'      => 'My\Namespaced\Suit',
+                'pureCases' => ['Hearts', 'Diamonds', 'Clubs', 'Spades'],
+            ],
+            <<<CODE
+                namespace My\Namespaced;
+                
+                enum Suit {
+                    case Hearts;
+                    case Diamonds;
+                    case Clubs;
+                    case Spades;
+                }
+                
+                CODE,
+        ];
+
+        yield 'backed string enums with namespace' => [
+            [
+                'name'        => 'My\Namespaced\Suit',
+                'backedCases' => [
+                    'type'  => 'string',
+                    'cases' => ['Hearts' => 'H', 'Diamonds' => 'D', 'Clubs' => 'C', 'Spades' => 'S'],
+                ],
+            ],
+            <<<CODE
+                namespace My\Namespaced;
+                
+                enum Suit: string {
+                    case Hearts = 'H';
+                    case Diamonds = 'D';
+                    case Clubs = 'C';
+                    case Spades = 'S';
+                }
+                
+                CODE,
+        ];
+
+        yield 'backed integer enums with namespace' => [
+            [
+                'name'        => 'My\Namespaced\Suit',
+                'backedCases' => [
+                    'type'  => 'int',
+                    'cases' => ['Hearts' => 1, 'Diamonds' => 2, 'Clubs' => 3, 'Spades' => 4],
+                ],
+            ],
+            <<<CODE
+                namespace My\Namespaced;
+                
+                enum Suit: int {
+                    case Hearts = 1;
+                    case Diamonds = 2;
+                    case Clubs = 3;
+                    case Spades = 4;
+                }
+                
+                CODE,
+        ];
+    }
+
+    public function testReflectionEnumFailsForUnsupportedPhpVersions(): void
+    {
+        if (PHP_VERSION_ID >= 80100) {
+            $this->markTestSkipped('ReflectionEnum is available from PHP 8.1 onwards.');
+        }
```

Comment:
You could use `/** @requires PHP >= 8.1 */` on test methods

Reply:
5e6e3d013c32b3681533143b8dc7bc27fcd25d1d